File: client/bcephal-client-model/Billing/Invoices/InvoiceDetail.cs
using Bcephal.Models.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Billing.Invoices
{
    public class InvoiceDetail : Persistent
    {

		public long? EventId { get; set; }

		public int Position { get; set; }

		public string Description { get; set; }

		public DateTime? Date { get; set; }

		public decimal? Quantity { get; set; }

		public string Unit { get; set; }

		public decimal? UnitCost { get; set; }

		public decimal? AmountWithoutVat { get; set; }

		public decimal? VatRate { get; set; }

		public string Currency { get; set; }


		public override int CompareTo(object obj)
		{
			if (obj == null || !(obj is InvoiceDetail)) return 1;
			if (this == obj) return 0;
			if (this.Id.HasValue && this.Id.Equals(((InvoiceDetail)obj).Id)) return 0;
			if (this.Position.Equals(((InvoiceDetail)obj).Position))
			{
				return this.Description.CompareTo(((InvoiceDetail)obj).Description);
			}
			return this.Position.CompareTo(((InvoiceDetail)obj).Position);
		}

		public override string ToString()
		{
			return this.Description;
		}

	}
}

File: server/bcephal-billing/src/main/java/com/moriset/bcephal/billing/domain/InvoiceDetail.java
/**
 * 
 */
package com.moriset.bcephal.billing.domain;

import java.math.BigDecimal;
import java.util.Date;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.moriset.bcephal.domain.Persistent;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * @author Joseph Wambo
 *
 */
@Entity(name = "InvoiceDetail")
@Table(name = "BCP_BILLING_INVOICE_DETAIL")
@Data
@EqualsAndHashCode(callSuper = false)
public class InvoiceDetail extends Persistent {

	private static final long serialVersionUID = 1389079201781359474L;
	
	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "billing_invoice_detail_seq")
	@SequenceGenerator(name = "billing_invoice_detail_seq", sequenceName = "billing_invoice_detail_seq", initialValue = 1,  allocationSize = 1)
	private Long id;

	@JsonIgnore
	private Long invoice;
	
	private Long eventId;
	
	private int position;
	
	private String description;
	
	private Date date;
	
	private BigDecimal quantity;
	
	private String unit;
	
	private BigDecimal unitCost;
	
	private BigDecimal amountWithoutVat;
		
	private BigDecimal vatRate;

	private String currency;


	@Override
	public Persistent copy() {
		return null;
	}
	
}
